editor/utils/dom.js

```diff
@@ -101,32 +97,47 @@ export function getVerticalEdge( container, isReverse ) {
 		: range.getClientRects()[ 0 ];
 
 	if ( ! rangeRect ) {
-		return { isEdge: false };
+		return false;
 	}
 
 	const buffer = rangeRect.height / 2;
 	const editableRect = container.getBoundingClientRect();
 
 	// Too low.
 	if ( isReverse && rangeRect.top - buffer > editableRect.top ) {
-		return {
-			rect: rangeRect,
-			isEdge: false,
-		};
+		return false;
 	}
 
 	// Too high.
 	if ( ! isReverse && rangeRect.bottom + buffer < editableRect.bottom ) {
-		return {
-			rect: rangeRect,
-			isEdge: false,
-		};
+		return false;
+	}
+
+	return true;
+}
+
+export function computeCaretRect( container ) {
+	if (
+		includes( [ 'INPUT', 'TEXTAREA' ], container.tagName ) ||
+		! container.isContentEditable
+	) {
+		return;
+	}
+
+	const selection = window.getSelection();
+	const range = selection.rangeCount ? selection.getRangeAt( 0 ) : null;
+
+	if ( ! range || ! range.collapsed ) {
+		return;
 	}
 
-	return {
-		rect: rangeRect,
-		isEdge: true,
-	};
+	// Adjust for empty containers.
+	const caretRect =
```

Comment:
Any reason to not return right away? :)

Reply:
Mostly vestigial :) but it seemed to make sense to pair the pre-existing comment with the const assignment. I can push a direct `return`.